File: leetcode/python/150_evaluate_reverse_polish_notation.py
class Solution:
    def evalRPN(self, tokens: list[str]) -> int:
        stack = []
        for token in tokens:
            if token in ["+", "-", "*", "/"]:
                b = stack.pop()
                a = stack.pop()
                if token == "+":
                    stack.append(a + b)
                elif token == "-":
                    stack.append(a - b)
                elif token == "*":
                    stack.append(a * b)
                elif token == "/":
                    stack.append(int(a / b))
            else:
                stack.append(int(token))
        return stack.pop()

File: leetcode/rust/src/150_evaluate_reverse_polish_notation.rs
impl Solution {
    pub fn eval_rpn(tokens: Vec<String>) -> i32 {
        let mut stack = vec![];
        for token in tokens {
            match token.as_str() {
                "+" | "-" | "*" | "/" => {
                    let b = stack.pop().unwrap();
                    let a = stack.pop().unwrap();
                    match token.as_str() {
                        "+" => stack.push(a + b),
                        "-" => stack.push(a - b),
                        "*" => stack.push(a * b),
                        "/" => stack.push(a / b),
                        _ => {}
                    }
                }
                _ => stack.push(token.parse::<i32>().unwrap()),
            }
        }
        stack[0]
    }
}